history/middleware.py

```diff
@@ -4,23 +4,33 @@
 
 from parsedatetime import parsedatetime, Constants
 from scrapy import signals
-from scrapy.xlib.pydispatch import dispatcher
 from scrapy.exceptions import NotConfigured, IgnoreRequest
 from scrapy.utils.misc import load_object
 
 MANDATORY_SETTINGS = ['HISTORY_S3_BUCKET',
                       'AWS_ACCESS_KEY_ID',
                       'AWS_SECRET_ACCESS_KEY']
+EPOCH_DATE_FORMAT = '%Y%m%d'
+
+
+def ignore_on_fail(func):
+    def _inner(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except Exception as e:
+            raise IgnoreRequest('middleware crashed: {err}'.format(err=e))
```

Comment:
On IgnoreRequest exception raising, will the rest of the stack still process the request, or will it be propagated to the other components (ie other Downloader middlewares, Spider Middlewares, Spider ...)? We need to process the response even if we can not store it on S3.
[From there](https://doc.scrapy.org/en/latest/topics/exceptions.html#ignorerequest) it looks like this request (and response) will be ignored for good so maybe we need to make sure it is not the case when testing.